ipu-plugin/pkg/ipuplugin/lifecycleservice.go

```diff
@@ -487,6 +489,102 @@ func (e *ExecutableHandlerImpl) validate() bool {
 	return true
 }
 
+func executeScript(script string) (string, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	cmd := exec.Command("sh", "-c", script)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("error calling sshFunc %s, %s, %v", stdout.String(), stderr.String(), err)
+	}
+	return stdout.String(), nil
+}
+
+/*
+* The AddFxpRules and DelFxpRules are two functions added as a workaround
+* to configure the FXP with point to point rules between the VFs initialised on a single
+* host. These rules assume that no NF has been deployed on the FXP.
+*
+* Function AddFxpRules will create all the point to point rules between all the initilised
+* VFs on the host.
+* Function DelFxpRules will remove all the point to point rules between all the initilised
+* VFs on the host.
+ */
+func AddFxpRules(p4rtbin string) {
+
+	// reach out to the IMC to get the mac addresses of the VFs
+	output, err := executeScript(`ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 root@192.168.0.1 "/usr/bin/cli_client -cq" \
+| awk '{if(($4 == "0x0") && ($6 == "yes")) {print $17}}'`)
+
+	if err != nil {
+		fmt.Printf("Unable to reach the IMC %v", err)
+		return
+	}
+
+	vfMacList := strings.Split(strings.TrimSpace(output), "\n")
+
+	for i := range vfMacList {
+		for j := range vfMacList {
+			if i != j {
+				vsi := strings.Trim(strings.Split(vfMacList[i], ":")[1], "\n")
+				targetVsi := strings.Trim(strings.Split(vfMacList[j], ":")[1], "\n")
+				vf2Port, err := strconv.ParseInt(targetVsi, 16, 64)
+
+				if err != nil {
+					fmt.Printf("error converting hex to decimal: %v", err)
+					return
+				}
+
+				vf2Port += 16
+				dmac := strings.Replace(vfMacList[j], string(':'), "", -1)
+
+				utils.RunP4rtCtlCommand(p4rtbin, "add-entry", "br0", "rh_mvp_control.ingress_loopback_table",
+					fmt.Sprintf("vsi=0x%s,target_vsi=0x%s,action=rh_mvp_control.fwd_to_port(%d)", vsi, targetVsi, vf2Port))
+				utils.RunP4rtCtlCommand(p4rtbin, "add-entry", "br0", "rh_mvp_control.vport_egress_dmac_vsi_table",
+					fmt.Sprintf("vsi=0x%s,dmac=0x%s,action=rh_mvp_control.fwd_to_port(%d)", vsi, dmac, vf2Port))
+			}
+		}
+	}
+}
+
+func DelFxpRules(p4rtbin string) {
+
+	// reach out to the IMC to get the mac addresses of the VFs
+	output, err := executeScript(`ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 root@192.168.0.1 "/usr/bin/cli_client -cq" \
+		| awk '{if(($4 == "0x0") && ($6 == "yes")) {print $17}}'`)
+
+	if err != nil {
+		fmt.Printf("Unable to reach the IMC %v", err)
+		return
+	}
+
+	vfMacList := strings.Split(strings.TrimSpace(output), "\n")
+
+	for i := range vfMacList {
```

Comment:
Can this for loop for parsing the mac and vsi info put into a function and then re-use for both in AddFxpRules and DelFxpRules? It looks quite similar other than the actual utils.RunP4rtCtlCommand.

Reply:
This code will be refactored once we add the code to generate the rules for network function comms. Will address  this comment in that PR.